schematic/synapse/store.py

```diff
@@ -65,12 +62,10 @@ def __init__(self,
             except synapseclient.core.exceptions.SynapseHTTPError:
                 print("Please enter a valid session token.")
                 return
-        elif syn: # if no token, assume a logged in synapseclient instance has been provided
-            if isinstance(syn, synapseclient.Synapse):
-                self.syn = syn
-            else:
-                print("Please make sure 'syn' argument is of type synapseclient.Synapse().")
-                return
+
+        # login using synapse credentials provided by user in .synapseConfig (default) file
```

Comment:
Maybe I am misinterpreting the logic here, but wouldn't self.syn be always set to read from Synapse config independent of whether a token was provided (L57) or not?

Is that the desired behavior? Wouldn't we still want to support token-based login when a token is provided?
Shouldn't l67-69 be in an else branch?

Reply:
Oh yes, good catch. I've moved the part that reads from the `.synapseConfig` file into an else-block.